test/build/50_dockerfile.bats

```diff
@@ -877,6 +877,39 @@ EOF
 }
 
 
+@test 'Dockerfile: FROM multistage alias' {
+    scope standard
+    run build_ --no-cache -t tmpimg -f - . <<'EOF'
+ARG BASEIMG=alpine:3.17
+FROM $BASEIMG as a
+RUN echo foo
+FROM a as b
+RUN echo bar
+FROM b
+RUN echo qux
+EOF
+    # We only care that other builders return 0; we only check ch-image output.
+    echo "$output"
+    [[ $status -eq 0 ]]
+    if [[ $CH_TEST_BUILDER = ch-image ]]; then
+        [[ $output = *"ARG BASEIMG='alpine:3.17'"* ]]
+        [[ $output = *'FROM alpine:3.17 AS a'* ]]
+        [[ $output = *'RUN echo foo'* ]]
+        [[ $output = *'FROM a AS b'* ]]
+        [[ $output = *'RUN echo bar'* ]]
+        [[ $output = *'FROM b'* ]]
+        [[ $output = *'RUN echo qux'* ]]
```

Comment:
Do we really need to validate all the instructions? If not, reduce to just the one you're concerned about; otherwise, comment to explain.